src/pilot/config_idrac.py

```diff
@@ -415,33 +427,48 @@ def config_idrac(instack_lock,
         ip_service_tag,
         node_definition)
 
-    # Configure the NIC port to PXE boot or not
-    reboot_required_nic, nic_job_ids, provisioning_mac = \
-        configure_nics_boot_settings(drac_client,
-                                     ip_service_tag,
-                                     pxe_nic_fqdd,
-                                     node,
-                                     target_boot_mode)
-
-    reboot_required = reboot_required or reboot_required_nic
-    if nic_job_ids:
-        job_ids.extend(nic_job_ids)
-
-    # Do initial idrac configuration
-    reboot_required_idrac, idrac_job_id = config_idrac_settings(drac_client,
-                                                                ip_service_tag,
-                                                                password,
-                                                                node)
-    reboot_required = reboot_required or reboot_required_idrac
-    if idrac_job_id:
-        job_ids.append(idrac_job_id)
-
-    # If we need to reboot, then add a job for it
-    if reboot_required:
-        LOG.info("Rebooting {} to apply configuration".format(ip_service_tag))
-
-        job_id = drac_client.create_reboot_job()
-        job_ids.append(job_id)
+    boot_modes_same = BootModeHelper.get_boot_mode(
+        drac_client) == target_boot_mode
+
+    if skip_nic_config and boot_modes_same:
```

Comment:
This story was originally created because an expert user configured the NICs manually and then wanted us to not change that configuration.  This was before UEFI support was added.  To meet the spirit of that story, I think we should NOT change the boot mode, but should leave it as configured when the user chooses to skip NIC configuration.  This is because the user could have configured the NICs as they wanted (how the PXE NIC is configured would be dependent on if the user pick UEFI or BIOS for the boot mode), but if we change the boot mode, that configuration will be lost.  As a result, I think we should change the code where target_boot_mode is initially set.  If skip_nic_configuration is specified, then we should just retrieve the boot mode from the server and set target_boot_mode to that.  This would make the boot_modes_same check unnecessary at this point in the code.

Reply:
done